frontend-v2/src/hooks/useDataset.ts

```diff
@@ -24,6 +25,12 @@ export default function useDataset(selectedProject: number | null) {
     { projectId: selectedProjectOrZero },
     { skip: !selectedProject },
   );
+  const { data: protocols } =
+    useProtocolListQuery(
+      {},
+      { skip: !selectedProject },
+    );
+  const datasetProtocols = protocols?.filter(p => parseInt(p.dataset) === dataset?.id);
```

Comment:
It would be nice to be able to request protcols by dataset ID and remove the array filter here. Something like:

```suggestion
  const { data: datasetProtocols } =
    useProtocolListQuery(
      { datasetId: dataset?.id },
      { skip: !selectedProject },
    );
```

Reply:
Fetching protocols by group ID might be useful too, for the tabs under Trial Design.